### PR TITLE
Rudimentary style customizability

### LINKED ISSUE #240: Rudimentary style customizability
We should offer the following styling options to make vaadin-grid a little easier to take into use:

- The blue highlight color should use `--default-primary-color` from paper-styles, and fallback to the same fixed color value if the custom property is not available/defined
- The row height should be exposed as a custom property, `--vaadin-grid-row-height`
- The font-family should be inherited from the parent element/page

All custom properties should be documented, and there should be a code example how to change the primary color and the row height.

### PR DESCRIPTION
**Pull request by tehapo**
Fixes #240

[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/259)

### CODE REVIEW

File: demo/sizing.html [hunk 1]
```diff
@@ -185,5 +185,106 @@
       </code>
     </code-example>
   </section>
+
+  <section>
+    <h3>Styling with Custom Properties</h3>
+    <p>The grid uses <code>--default-primary-color</code> from paper-styles as a highlight color.</p>
```
Comment (jounik) [inline demo/sizing.html]: Could have a link to the paper-styles GitHub repo, since some might not know what that is.

**Comment by jounik** [comment]
LGTM

**Comment by tehapo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[demo/sizing.html, line 191 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259#-K3--B4TioKIn3f7kD0B-r1-191)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/b6fb6abb0d333e5b640d6514a37f523f890f922e/demo/sizing.html#L191)):</sup>
Done. Added a link to paper-styles GitHub repo here and also to the apidocs.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259)

**Comment by tomivirkki** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

<sup>**[demo/sizing.html, line 194 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259#-K3JWqKlBCeYzlGn_28k)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/c2ed20f0bb19dadda5c2b3ad9d082153a513cefd/demo/sizing.html#L194)):</sup>
You can use <code-example> here also. Just include the escaped <style> section inside the  <code> block, and wrap it in /* .... */ so it doesn't get evaluated. Then start the "// code" right after /*. The same approach was used with angular2.html to include typescript in the code-example.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259)

**Comment by tehapo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

<sup>**[demo/sizing.html, line 194 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259#-K3JWqKlBCeYzlGn_28k)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/c2ed20f0bb19dadda5c2b3ad9d082153a513cefd/demo/sizing.html#L194)):</sup>
Done. Thanks for the info, got rid of the custom example.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259)

**Comment by Saulis** [comment]
Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[vaadin-grid.html, line 68 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259#-K3JkmzZRpipjQXCIRr2)** ([raw file](https://github.com/vaadin/vaadin-grid/blob/d124ff00b2cd5eba9f41e11a7b12fde9c2ceb12e/vaadin-grid.html#L68)):</sup>
Should this be mentioned somewhere in the examples or docs?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259)

**Comment by tehapo** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_[vaadin-grid.html, line 68 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259#-K3JkmzZRpipjQXCIRr2:-K3KIPjRGy6XyL8fpnNM:-392493616) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d124ff00b2cd5eba9f41e11a7b12fde9c2ceb12e/vaadin-grid.html#L68)):_
This was intentionally left out of examples and docs as it was thought to be a "default behavior".

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259)_

**Comment by tomivirkki** [comment]
LGTM

---







_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/259#-:-K3KJH7j9XuS1bcoFcJl:676834233)_

**Pull request merged**